docs/proposal/kmesh_support_localityLB.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Locality Load Balancing for Kmesh
+authors:
+- "@derekwin" # Authors' GitHub accounts here.
+reviewers:
+- "@kwb0523"
+- "@hzxuzhonghu"
+approvers:
+- "@robot"
+- TBD
+
+creation-date: 2024-06-07
+
+---
+
+## Locality Load Balancing for Kmesh
+
+### Summary
+
+Add Locality Load Balancing to Kmesh.
+
+### Motivation
+
+Currently, kmesh does not support locality topology-aware load balancing. Locality Load Balancing optimizes performance and reliability in distributed systems by directing traffic to the nearest service instances. This reduces latency, enhances availability, and lowers costs associated with cross-region data transfers. It also helps ensure compliance with data sovereignty regulations and improves overall user experience by providing faster and more reliable service responses.
+
+#### Goals
+
+The purpose of this proposal is to add locality-based topology-aware load balancing capabilities to kmesh, corresponding to locality load balancing in Istio.
+
+### Proposal
+
+1. prioritize add locality load balancing capabilities in the workload mode.
+
+2. two types of locality load balancing : locality failover, locality weighted distribution.
+
+#### case 1. locality failover
+1. Destination Rule
+Same as Istion. Parse rules specify configuration for Locality load balancing. (todo: outlier detection settings to detect and evict unhealthy hosts from the load balancing pool.)
+    - Outlier detection should occur before load balancing.
```

Comment:
This does not suite worklaod mode as workload api does not include outlier setting. It do LB based on where the endpoint resides.